webknossos-tracingstore/app/com/scalableminds/webknossos/tracingstore/controllers/VolumeTracingController.scala

```diff
@@ -395,28 +398,35 @@ class VolumeTracingController @Inject()(
   private def getFallbackLayerDataIfEmpty(tracing: VolumeTracing,
                                           data: Array[Byte],
                                           missingBucketIndices: List[Int],
-                                          mag: String,
-                                          cxyz: String,
+                                          mag: Vec3Int,
+                                          position: Vec3Int,
+                                          cubeSize: Int,
                                           urlToken: Option[String]): Fox[Array[Byte]] = {
-
     def fallbackLayerData(): Fox[Array[Byte]] = {
       val organizationName = tracing.organizationName
       val dataSetName = tracing.dataSetName
       val dataLayerName = tracing.getFallbackLayer
+      val request = WebKnossosDataRequest(
+        position = Vec3Int(position.x * cubeSize * mag.x, position.y * cubeSize * mag.y, position.z * cubeSize * mag.z),
+        mag = mag,
+        cubeSize = cubeSize,
+        fourBit = Some(false),
+        applyAgglomerate = tracing.mappingName,
+        version = None
+      )
 
       organizationName match {
         case Some(orgName) =>
           for {
-            fallbackData <- remoteDataStoreClient.fallbackLayerBucket(
+            // ignoring missing bucket indices, if data is empty overall response is empty anyways
+            (fallbackData, _) <- remoteDataStoreClient.getData(
```

Comment:
will the response be an empty array, or a Fox.empty/Fox.failure? I think an empty array would not currently produce a 404, is this right :thinking: ?

Reply:
You are right, I adapted the code to return a 404 when the data is not available in the fallback layer